flow/connectors/mongo/mongo.go

```diff
@@ -404,26 +422,60 @@ func (c *MongoConnector) PullRecords(
 	return nil
 }
 
-func defaultPipeline() mongo.Pipeline {
-	return mongo.Pipeline{
-		{{Key: "$match", Value: bson.D{
+func createPipeline(tableNameMapping map[string]model.NameAndExclude) (mongo.Pipeline, error) {
+	pipeline := mongo.Pipeline{}
+
+	// filter out events from tables that are not in the mapping
+	if tableNameMapping != nil {
+		dbCollMap := make(map[string][]string)
+		for dbAndTable := range tableNameMapping {
+			parts := strings.SplitN(dbAndTable, ".", 2)
```

Comment:
not related to this PR but wondering how we handle collections/tables with a period in the name